### PR TITLE
Add missing HTTPS configs

### PR DESCRIPTION
**Pull request by Darrekt**
While trying to follow the [official instructions for TLS/HTTPS](https://trino.io/docs/current/security/tls.html#secure-trino-directly), I realised that a lot of the required flags in `config.properties` for the coordinator were not yet implemented.

This PR attempts to implement them to the best of my knowledge. This is my very first open-source contribution, so if I did something wrong I am happy to receive feedback and fix my mistakes / convention breaks accordingly.

The following changes were made:
- `secretMounts` was incorrectly implemented: In `deployment-coordinator.yaml`, the `initContainers` clause was incorrectly in the middle of the `volumes` clause, interrupting the place where the `secretMounts` would be added. I've fixed this to permit correct volume addition for certificates which I assume would be mounted using this mechanism.
- `keystore.keystorePassword` was previously not implemented to support keystores with passwords.
- `keystore.keymanagerPassword` was similarly missing.
- Added support for the `allowInsecureOverHttp` flag.

Edit: Would also like some help on getting around the CLA checks. I've submitted my signed CLA a few days ago, and signed my commits with my email using `git rebase --signoff HEAD~2` after setting my email in my git client, but still getting the error.

### CODE REVIEW

**Comment by Darrekt** [comment]
@cla-bot check

**Comment by Darrekt** [comment]
Yep. I'm happy to integrate changes if I get some feedback. Also, some help on how to appease the cla-bot would be appreciated. I've signed my commits, but it's still complaining.

**Comment by Torugo** [comment]
I'm having some problems when trying have an https connection to my cluster as well, how did you managed to do it? I'm also unfamiliar with kubernets

File: charts/trino/values.yaml [hunk 1]
```diff
@@ -45,7 +48,8 @@ server:
     maxReplicas: 5
     targetCPUUtilizationPercentage: 50
 
-accessControl: {}
+accessControl:
```
Comment (nineinchnick) [inline charts/trino/values.yaml:51]: Revert these changes, they're not related to HTTPS improvements.

File: charts/trino/values.yaml [hunk 2]
```diff
@@ -24,11 +24,14 @@ server:
     path: /etc/trino
     http:
       port: 8080
+      allowInsecureOverHttp: false
```
Comment (nineinchnick) [inline charts/trino/values.yaml:27]: This can be added in `server.coordinatorExtraConfig` or `additionalConfigProperties`. We should not encourage using this, as it reduces the security, so I'm not sure if we want a dedicated value for this. If you disagree, can you open a separate PR with this change?

File: charts/trino/templates/deployment-coordinator.yaml [hunk 3]
```diff
@@ -47,15 +47,15 @@ spec:
...
-      {{- if .Values.initContainers.coordinator }}
-      initContainers:
-      {{-  tpl (toYaml .Values.initContainers.coordinator) . | nindent 6 }}
-      {{- end }}
         {{- range .Values.secretMounts }}
         - name: {{ .name }}
           secret:
             secretName: {{ .secretName }}
         {{- end }}
+      {{- if .Values.initContainers.coordinator }}
```
Comment (nineinchnick) [inline charts/trino/templates/deployment-coordinator.yaml:55]: Please rebase, I think this should have been fixed already.

**Comment by nineinchnick** [comment]
Adding the missing properties is a good change, but this PR is abandoned, so I'm closing it.
